ledger/participant-integration-api/src/test/suite/scala/platform/indexer/parallel/ParallelIndexerFactorySpec.scala

```diff
@@ -0,0 +1,308 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.platform.indexer.parallel
+
+import akka.stream.KillSwitch
+import com.daml.ledger.api.testing.utils.AkkaBeforeAndAfterAll
+import com.daml.ledger.resources.{Resource, ResourceContext, ResourceOwner}
+import com.daml.platform.indexer.ha.Handle
+import org.scalatest.flatspec.AsyncFlatSpec
+import org.scalatest.matchers.should.Matchers
+
+import scala.concurrent.{ExecutionContext, Future, Promise}
+
+class ParallelIndexerFactorySpec extends AsyncFlatSpec with Matchers with AkkaBeforeAndAfterAll {
+
+  // AsyncFlatSpec is with serial execution context
+  private implicit val ec: ExecutionContext = system.dispatcher
+
+  behavior of "initializeHandle"
+
+  it should "correctly chain initializations and teardown-steps in the happy path" in {
+    val t = test
+    import t._
+
+    resourceReleasing.isCompleted shouldBe false
+    initialized.isCompleted shouldBe false
+    initHandleStarted.isCompleted shouldBe false
+
+    Thread.sleep(10)
+
+    resourceReleasing.isCompleted shouldBe false
+    initialized.isCompleted shouldBe false
+    initHandleStarted.isCompleted shouldBe false
+
+    resourceInitPromise.success("happy")
+
+    val completePromise = Promise[Unit]()
+
+    for {
+      s <- initHandleStarted
+      _ = {
+        Thread.sleep(10)
```

Comment:
Why do we need those sleeps? s should be "happy" at this point without a sleep as the `initHandleStarted` future has completed 2 lines above.

Reply:
"wait" for something else to happen

Reply:
I'll add a comment